src/support/metadata/TypeMetadata.ts

```diff
@@ -2,6 +2,37 @@ export class TypeMetadata {
     constructor(public target: Function,
                 public propertyName: string,
                 public sourcePropertyName: string,
-                public type: Function) {
+                private _type: Function) {
+    }
+
+    public get type(): Function
+    {
+        // If type name is empty, we will assume it returns a 'require'.
+        if (!this._type.name.length) {
+            return this.resolveDeferredType();
+        }
+
+        return this._type;
+    }
+
+    private resolveDeferredType(): Function
+    {
+        // Run the function to actually import the module and assign the module
+        // to type prop so that the EntityBuilder will actually get an entity
+        // constructor, and not a Promise.
```

Comment:
I'm guessing this should say `and not a callback` rather than `and not a Promise`